### ISSUE TITLE
Update DragLayer style based on which DropTarget is hovered

### ISSUE THREAD

**Issue by vladbicu** [react-dnd/react-dnd#1562]
Hello all, 

I'm working at an app that has a couple of `DragSources` with different types and a couple of `DropTargets` that can accept or not the `DragSources`, based on `accept` array. When a drag is performed I added a custom `DragLayer`.

When the `DragLayer` is over a `DropTarget` i want to change the style of `DragLayer`(add a green border if it can be dropped or a red one if it cannot be dropped). 

It is a solution(just using react-dnd) to communicate between a `DropTarget` and a `DragLayer`? How I should tackle this problem?

**Comment by anmlkh** [comment]
I have similar task. I think it would be nice to have next info from `DropTargetMonitor` in `DragLayerMonitor`:
- isOver
- canDrop

**Comment by vladbicu** [comment]
@anmlkh agree. I was able to pass the mentioned information between `DropTarget` and `DragLayer` using callback props. The callback function is called in `DropTarget`:

`useEffect(() => {
    if (isOverCurrent) {
        callback(canDrop)
    }
})`

How did you solve the problem?

**Comment by yizhengfeng-jj** [comment]
@vladbicu  Hello, I think I can help you solve your problem

When we print monitor in the collection function of draglayer, we will find that there are two functions: gettargetids and isovertarget. Isovertarget is to judge whether it is on the specified drop layer. See the following print.
![image](https://user-images.githubusercontent.com/20452750/67174041-34cd8f00-f3f3-11e9-83c9-1490b8361d84.png)

## Be careful
We have seen that isovertarget needs an ID, so where does this ID come from? It is obtained from gettargetids. But you will find that your print moniter. Gettargetids() is an empty array. Don't worry. This is because you haven't moved into the drop layer yet. When you move into the drop layer, you will see the printed ID. At this time, the ID is an array, so you need to cycle. If the ID has only one element, you can use the ID [0] directly.

## critical code
![image](https://user-images.githubusercontent.com/20452750/67174653-6b0c0e00-f3f5-11e9-807a-5d9169d0461d.png)


This is a complete code example https://codesandbox.io/s/angry-star-vrdmb I hope I can help you

**Comment by williamtstanley** [comment]
When searching to try to solve the same problem I found this thread. I am attempting something similar however I am using the hooks versions.

The monitor provided by useDragLayer doesn't have the functions `getTargetIds` and `isOverTarget` exposed. Is there a Way to accomplish the same result or am I out of luck?

At the moment I am faking it with `document.elementFromPoint` and some data-attrs. It would be way simpler if I could access target somehow in the collect function.

Thanks in advance
